app/backend/src/couchers/servicers/donations.py

```
@@ -139,34 +121,34 @@ def Webhook(self, request, context, session):
         event_type = event["type"]
         event_id = event["id"]
         data_object = data["object"]
+        metadata = data_object.get("metadata", {})
 
         # Get the type of webhook event sent - used to check the status of PaymentIntents.
         logger.info(f"Got signed Stripe webhook, {event_type=}, {event_id=}")
 
         if event_type == "charge.succeeded":
-            customer_id = data_object["customer"]
-            user = session.execute(select(User).where(User.stripe_customer_id == customer_id)).scalar_one()
-            # amount comes in cents
-            amount = int(data_object["amount"]) // 100
-            receipt_url = data_object["receipt_url"]
-            payment_intent_id = data_object["payment_intent"]
-
-            # Get invoice type from charge metadata
-            invoice_type = infer_invoice_type(data_object.get("metadata", {}), payment_intent_id, customer_id)
-
-            invoice = Invoice(
-                user_id=user.id,
-                amount=amount,
-                stripe_payment_intent_id=payment_intent_id,
-                stripe_receipt_url=receipt_url,
-                invoice_type=invoice_type,
-            )
-            session.add(invoice)
-            session.flush()
-            user.last_donated = invoice.created
+            if metadata.get("site_url") == config["MERCH_SHOP_URL"]:
+                # merch shop. just look up this email and update their last donated
+                session.execute(update(User).where(User.email == metadata["customer_email"]).values(last_donated=now()))
```

If customer_email is not there, this will fail, Stripe will resend the webhook, and we'll be in a loop (and will lose the hook eventually)

---

yes but it'll raise an error and we'll find out, eventually stripe will stop :)

good to fail loudly. i mean what do we do with the webhook otherwise anyway?